src/app/(tools)/coverLetter/page.tsx

```diff
@@ -0,0 +1,285 @@
+"use client";
+
+import { useState, useRef } from 'react';
+import { Upload, X } from 'lucide-react';
+import axios from 'axios';
+import jsPDF from 'jspdf';
+import { useForm, Controller } from 'react-hook-form';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+
+import { Button } from "@/components/ui/button";
+import { Textarea } from "@/components/ui/textarea";
+import { toast } from '@/hooks/use-toast';
+
+const schema = z.object({
+  jobDescription: z.string().min(1, "Job description cannot be empty"),
+  resume: z.string().min(1, "Resume cannot be empty")
+});
+
+type FormData = z.infer<typeof schema>;
+
+export default function Page() {
+  const [coverLetter, setCoverLetter] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [uploadedFileName, setUploadedFileName] = useState<string | null>(null);
+  const fileInputRef = useRef<HTMLInputElement>(null);
+  const coverLetterRef = useRef<HTMLTextAreaElement>(null);
+
+  const { control, handleSubmit, setValue, formState: { errors } } = useForm<FormData>({
+    resolver: zodResolver(schema),
+    defaultValues: {
+      jobDescription: '',
+      resume: ''
+    }
+  });
+
+  const handleResumeChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setValue('resume', e.target.value);
+  };
+
+  const handleResumePaste = (e: React.ClipboardEvent) => {
+    const items = e.clipboardData.items;
+    for (let i = 0; i < items.length; i++) {
+      if (items[i].kind === 'file') {
+        e.preventDefault();
+        const file = items[i].getAsFile();
+        if (file) handleFileUpload(file);
+        break;
+      }
+    }
+  };
+
+  const handleFileUpload = async (file: File) => {
```

Comment:
this could have been cleaner and resuable as a customHook "useFileUpload" 

learn how to create custom hooks, once you get used to it, it will become muscle memory and code will become a lot cleaner. they will be your best friend.

ask AI to make it into a custom hook and see what it does.

Reply:
lets say in future, the requirement is to persist the uploaded file URLs in local storage. then the only place you need to update code will be in useFileUpload hook and will reflect across all the tools.